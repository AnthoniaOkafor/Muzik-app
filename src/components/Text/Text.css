.green {
    color: green;
}

.white {
    color: white;
}

.small {
    font-size: 20px;
}

.medium {
    font-size: 32px;
}

.large {
    font-size: 40px;
}

