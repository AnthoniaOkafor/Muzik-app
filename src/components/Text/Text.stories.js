import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { withKnobs, text } from "@storybook/addon-knobs";
import Text from './index';


storiesOf('Text', module)
	.addDecorator(
		//Allows you to provide information about the component
		withInfo({
			text: "This is a text component"
		})
	)
	.addDecorator(
		withKnobs() //allows you to interact with the component in Storybook
	)
	.add('Text', () => (
		<Text text={text('text', 'Sample text')} />
	));