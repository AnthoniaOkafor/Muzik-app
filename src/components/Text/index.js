
import React from "react";
import './Text.css';

//<Text text='Sample text' color='green' size="small" />
const Text = (props) => {
  const { text, color, size } = props;
  //size=['small', 'medium', 'large']
  //        20px      32px      40px
  
  const className = `${color} ${size}`; // 'green small'
  return (
    <p className={className}>{text}</p>
  )
}

export default Text;