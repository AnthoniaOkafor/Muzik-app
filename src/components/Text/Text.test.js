import React from "react"
import Text from './index'
import {shallow} from 'enzyme'

describe ('Text Component render', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<Text text='It is a text' />);
        expect(wrapper).toMatchSnapshot();
    });

    it('displays text', () => {
        const wrapper = shallow(<Text text='It is a text' />);
        expect(wrapper.text()).toBe('It is a text');
    });
});