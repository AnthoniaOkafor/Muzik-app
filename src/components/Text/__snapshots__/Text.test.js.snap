// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Component render renders correctly 1`] = `ShallowWrapper {}`;
